pkg/subctl/cmd/join.go

```diff
@@ -242,13 +243,23 @@ func joinSubmarinerCluster(config *rest.Config, subctlData *datafile.SubctlData)
 	status.Start("Discovering multi cluster details")
 	globalNetworks := getGlobalNetworks(subctlData)
 
-	err = checkOverlappingServiceCidr(globalNetworks)
-	status.End(err == nil)
-	exitOnError("Error validating overlapping ServiceCIDRs", err)
-
-	err = checkOverlappingClusterCidr(globalNetworks)
-	status.End(err == nil)
-	exitOnError("Error validating overlapping ClusterCIDRs", err)
+	if subctlData.GlobalnetCidrRange == "" {
+		// Globalnet not enabled
+		err = checkOverlappingServiceCidr(globalNetworks)
+		status.End(err == nil)
+		exitOnError("Error validating overlapping ServiceCIDRs", err)
+		err = checkOverlappingClusterCidr(globalNetworks)
+		status.End(err == nil)
+		exitOnError("Error validating overlapping ClusterCIDRs", err)
+	} else if globalNetworks[clusterID] == nil || globalNetworks[clusterID].GlobalCIDRs == nil || len(globalNetworks[clusterID].GlobalCIDRs) <= 0 {
+		globalCIDR, err = globalnet.AllocateGlobalCIDR(globalNetworks, subctlData)
+		status.End(err == nil)
+		status.QueueSuccessMessage(fmt.Sprintf("Allocated GlobalCIDR: %s", globalCIDR))
+		exitOnError("Globalnet failed", err)
+	} else {
+		globalCIDR = globalNetworks[clusterID].GlobalCIDRs[0]
+		status.QueueSuccessMessage(fmt.Sprintf("Cluster already has GlobalCIDR allocated: ", globalNetworks[clusterID].GlobalCIDRs[0]))
```

Comment:
%s missing in the printf message is causing build failures

Reply:
Done.